src/test_utils/kbn_server.js

```diff
@@ -92,9 +95,12 @@ export function authOptions() {
  *
  * @param {KbnServer} kbnServer
  * @param {object}    options Any additional options or overrides for inject()
- * @param {Function}  fn The callback to pass as the second arg to inject()
  */
-export function makeRequest(kbnServer, options, fn) {
-  options = defaultsDeep({}, authOptions(), options);
-  return kbnServer.server.inject(options, fn);
+export function makeRequest(kbnServer, options) {
+  // Since all requests to Kibana hit core http server first and only after that
+  // are proxied to the "legacy" Kibana we should inject requests through the top
+  // level Hapi server used by the core.
+  return kbnServer.newPlatform.proxyListener.root.server.http.service.httpServer.server.inject(
```

Comment:
😨 Maybe `kbnServer.newPlatform.proxyListener` could expose a method that is a little less... hacky?

Reply:
I can, but then we'll have hacky stuff in non-test code since most of these nested properties are private in TS as they should be. Aaaand I'll get rid of `kbnServer.newPlatform` entirely in 2-3 PRs :) Or you have some specific suggestion in mind already?

Reply:
Yep, I have _some_ plan, still in flux though, but I'll see what I can come up with in the scope of https://github.com/elastic/kibana/pull/19994.